bowling/src/frameParser.cpp

```diff
@@ -0,0 +1,72 @@
+#include <algorithm>
+#include <functional>
+
+#include "frameParser.hpp"
+#include "tokenizer.hpp"
+
+const char FrameParser::StrikeSign = 'X';
+const char FrameParser::MissSign = '-';
+const char FrameParser::SpareSign = '/';
+
+void FrameParser::parseToken(const char token, size_t& lastRollValue) {
+    switch (token) {
+    case FrameParser::StrikeSign:
+        lastRollValue = 10;
+        break;
+    case FrameParser::SpareSign:
+        lastRollValue = 10 - lastRollValue;
+        break;
+    case FrameParser::MissSign:
+        lastRollValue = 0;
+        break;
+    default:
+        lastRollValue = static_cast<size_t>(token - '0');
+        break;
+    }
+}
+
+parsedFrame FrameParser::parse(const std::string& line) {
+    std::vector<size_t> parsed{};
+    size_t lastRollValue{};
+
+    auto sequence = Tokenizer::readSequence(line);
+
+    std::for_each(sequence.cbegin(), sequence.cend(), [&lastRollValue, &parsed](const auto& el) {
+        FrameParser::parseToken(el, lastRollValue);
+        parsed.push_back(lastRollValue);
+    });
+
+    return parsed;
+}
+
+bool FrameParser::isSequenceComplete(const std::string& line) {
+    const std::string endOfSequenceDelimiter = "||";
+
+    size_t endOfSequenceDelimiterPosition = line.find(endOfSequenceDelimiter);
+
+    if (endOfSequenceDelimiterPosition == std::string::npos) {
+        return false;
+    } else {
+        auto lastRegularToken = *(std::prev((line.cbegin() + endOfSequenceDelimiterPosition)));
+        size_t numberOfTokensToLookFor;
+        if (lastRegularToken == FrameParser::StrikeSign) {
+            numberOfTokensToLookFor = 2;
+            return areProperTokensAtTheEnd(line, endOfSequenceDelimiterPosition, numberOfTokensToLookFor);
+        } else if (lastRegularToken == FrameParser::SpareSign) {
+            numberOfTokensToLookFor = 1;
+            return areProperTokensAtTheEnd(line, endOfSequenceDelimiterPosition, numberOfTokensToLookFor);
+        } else {
+            return true;
+        }
+    }
```

Comment:
```suggestion
    if (endOfSequenceDelimiterPosition == std::string::npos) {
        return false;
    }
    auto lastRegularToken = *(std::prev((line.cbegin() + endOfSequenceDelimiterPosition)));
    size_t numberOfTokensToLookFor;
    if (lastRegularToken == FrameParser::StrikeSign) {
        numberOfTokensToLookFor = 2;
        return areProperTokensAtTheEnd(line, endOfSequenceDelimiterPosition, numberOfTokensToLookFor);
    }
    if (lastRegularToken == FrameParser::SpareSign) {
        numberOfTokensToLookFor = 1;
        return areProperTokensAtTheEnd(line, endOfSequenceDelimiterPosition, numberOfTokensToLookFor);
    }
    return true;
```

Reply:
Since return escapes function elses are redundant. You can rewrite this fragment similar to way I suggested.